packages/flutter/lib/src/cupertino/segmented_control.dart

```diff
@@ -0,0 +1,683 @@
+// Copyright 2018 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/foundation.dart';
+import 'package:flutter/rendering.dart';
+import 'package:flutter/widgets.dart';
+
+import 'colors.dart';
+
+// Minimum padding from horizontal edges of segmented control to edges of
+// encompassing widget.
+const EdgeInsets _kHorizontalItemPadding = const EdgeInsets.symmetric(horizontal: 16.0);
+
+// Minimum height of the segmented control.
+const double _kMinSegmentedControlHeight = 28.0;
+
+// Light blue color. Used to fill the background of a child option the user
+// is temporarily interacting with through a long press or drag.
+const Color _kPressedBackground = const Color(0x33007aff);
+
+/// An iOS-style segmented control.
+///
+/// Displays the provided [children] widgets in a horizontal list.
+/// Used to select between a number of mutually exclusive options.
+/// When one option in the segmented control is selected, the other options
+/// in the segmented control cease to be selected.
+///
+/// A segmented control can feature any [Widget] as one of its [children].
+/// All [children] must have a [Key] value that can be used to identify
+/// the widget. Typically, these keys will be instances of the [ValueKey]
+/// class.
+///
+/// When the state of the segmented control changes, the widget calls the
+/// [onValueChanged] callback. The [Key] of the currently selected widget
+/// is returned in the [onValueChanged] callback. Most widgets that use
+/// a segmented control will listen for the [onValueChanged] callback
+/// and rebuild the segmented control with a new [value] to update which
+/// option is currently selected.
+///
+/// The height of the segmented control is determined by the height of the
+/// tallest widget provided in the [children] list. The width of the segmented
+/// control is determined by the horizontal constraints on its parent. The
+/// available horizontal space is divided by the number of provided [children]
+/// to determine the width of each widget. The selection area for each of the
+/// [children] will then  be expanded to fill the calculated space, so each
+/// widget will appear to have the same dimensions.
+///
+/// See also:
+///
+/// * <https://developer.apple.com/design/human-interface-guidelines/ios/controls/segmented-controls/>
```

Comment:
super minor nit: lists should be indented by one (two spaces between `///` and `*`)